### PR TITLE
unfurl links without preview

### PR DESCRIPTION
**Pull request by prsvic**
This pull request adds logic for unfurling blinks that do not have a link preview.

- If a tweet has a link preview, everything works as before. Even if the tweet has another blink URL, only the one with a corresponding link preview will unfurl.

- If a tweet does not have a link preview, the last URL in the tweet text will unfurl.

### CODE REVIEW

File: src/ui/ActionLayout.tsx [hunk 1]
```diff
@@ -119,13 +119,13 @@ export const ActionLayout = ({
...
                 <LinkIcon className="mr-2 text-icon-primary transition-colors group-hover:text-icon-primary-hover motion-reduce:transition-none" />
-                <span className="text-text-link group-hover:text-text-link-hover transition-colors group-hover:underline motion-reduce:transition-none">
+                <span className="text-text-link transition-colors group-hover:text-text-link-hover group-hover:underline motion-reduce:transition-none">
                   {websiteText ?? websiteUrl}
                 </span>
               </a>
             )}
             {websiteText && !websiteUrl && (
-              <span className="text-text-link -mt-1 inline-flex items-center truncate text-subtext">
+              <span className="-mt-1 inline-flex items-center truncate text-subtext text-text-link">
```
Comment (fsher) [inline src/ui/ActionLayout.tsx:128]: can you please these offsets as well pls `-mt-1`: here and on line 118

File: src/ext/twitter.tsx [hunk 2]
```diff
@@ -242,3 +256,61 @@ function findElementByTestId(element: Element, testId: string) {
...
+function findContainerInTweet(element: Element, searchUp?: boolean) {
+  const message = searchUp
+    ? element.closest(`[data-testid="tweet"]`) ??
+      element.closest(`[data-testid="messageEntry"]`)
+    : findElementByTestId(element, 'tweet') ??
+      findElementByTestId(element, 'messageEntry');
+
+  if (message) {
+    return message.querySelector('.dialect-wrapper') as HTMLElement;
+  }
```
Comment (fsher) [inline src/ext/twitter.tsx:269]: minor: please add `return null` here, not a huge fan of implicit returns

File: src/ext/twitter.tsx [hunk 3]
```diff
@@ -242,3 +256,61 @@ function findElementByTestId(element: Element, testId: string) {
...
+}
+
+function findLinkPreview(element: Element) {
+  const card = findElementByTestId(element, 'card.wrapper');
+  if (card) {
```
Comment (fsher) [inline src/ext/twitter.tsx:274]:
```ts
if (!card) {
  return null;
}

const anchor = card.children[0]?.children[0];

return anchor ? { anchor, card } : null;
```

File: src/ext/twitter.tsx [hunk 4]
```diff
@@ -242,3 +256,61 @@ function findElementByTestId(element: Element, testId: string) {
...
+  const tweetText = findElementByTestId(element, 'tweetText');
+  if (!tweetText) {
+    return;
+  }
+
+  const links = tweetText.getElementsByTagName('a');
+  if (links.length > 0) {
+    const anchor = links[links.length - 1] as HTMLAnchorElement;
+    return { anchor, tweetText };
+  }
```
Comment (fsher) [inline src/ext/twitter.tsx:292]: return null

File: src/ui/ActionLayout.tsx [hunk 5]
```diff
@@ -96,7 +96,7 @@ export const ActionLayout = ({
 }: LayoutProps) => {
   return (
     <div className={clsx('blink', stylePresetClassMap[stylePreset])}>
-      <div className="mt-3 w-full cursor-default overflow-hidden rounded-2xl border border-stroke-primary bg-bg-primary shadow-action">
+      <div className="w-full cursor-default overflow-hidden rounded-2xl border border-stroke-primary bg-bg-primary shadow-action">
```
Comment (fsher) [inline src/ui/ActionLayout.tsx:99]: thank you.... i wanted to do this as well

**Review by fsher — APPROVED** [review]
haven't ran it yet, but the code looks good to me

File: src/ext/twitter.tsx [hunk 6]
```diff
@@ -124,7 +124,7 @@ async function handleNewNode(
   );
   if (linkPreview) {
     anchor = linkPreview.anchor;
-    container?.remove();
+    container && container.remove();
```
Comment (zhelezkov) [inline src/ext/twitter.tsx:127]: `container?.remove();` was better

**Pull request merged**